frontend/src/app/main/ui/workspace/viewport/scroll_bars.cljs

```diff
@@ -0,0 +1,139 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) UXBOX Labs SL
+
+(ns app.main.ui.workspace.viewport.scroll-bars
+  (:require
+   [app.common.uuid :as uuid]
+   [app.common.geom.shapes :as gsh]
+   [app.common.geom.shapes.rect :as gpr]
+   [app.main.data.workspace :as dw]
+   [app.main.refs :as refs]
+   [app.main.store :as st]
+   [app.main.ui.workspace.viewport.utils :as utils]
+   [app.util.dom :as dom]
+   [rumext.alpha :as mf]))
+
+(mf/defc viewport-vertical-scrollbar
+  {::mf/wrap [mf/memo]}
+  [{:keys [viewport-ref zoom vbox]}]
+
+  (let [scrolling?              (mf/use-state false)
+        start-ref               (mf/use-ref nil)
+        scrollbar-y-ref         (mf/use-ref nil)
+        scrollbar-y-stored      (mf/ref-val scrollbar-y-ref)
+        scrollbar-y-padding-ref (mf/use-ref nil)
+        scrollbar-height-ref    (mf/use-ref nil)
+        scrollbar-height-stored (mf/ref-val scrollbar-height-ref)
+        height-factor-ref       (mf/use-ref nil)
+        vbox-y-ref              (mf/use-ref nil)
+
+        base-objects            (mf/deref refs/workspace-page-objects)
+        base-objects-rect
+        (mf/use-memo
+         (mf/deps base-objects)
+         (fn []
+           (let [root-shapes             (get-in base-objects [uuid/zero :shapes])
+                 shapes                  (->> root-shapes (mapv #(get base-objects %)))]
+             (gsh/selection-rect shapes))))
+
+        inv-zoom                (/ 1 zoom)
+        vbox-height             (- (:height vbox) (* inv-zoom 60)) 
+
+        ;; top space hidden because of the scroll
+        top-offset              (-> (- (:y vbox) (:y base-objects-rect))
+                                    (max 0)
+                                    (* vbox-height)
+                                    (/ (:height base-objects-rect)))
+
+        ;; bottom space hidden because of the scroll
+        bottom-offset           (-> (- (:y2 base-objects-rect) (+ (:y vbox) vbox-height))
+                                    (max 0)
+                                    (* vbox-height)
+                                    (/ (:height base-objects-rect)))
+
+        show-vertical-scroll?   (or @scrolling? (> top-offset 0) (> bottom-offset 0))
+
+        scrollbar-x             (+ (:x vbox) (:width vbox) (* inv-zoom -32))
```

Comment:
for performance reasons, consider extract the commonly used attrs from `vbox` such that `x` and `y` (they are used many times as `(:x vbox)...`, this improves performance because the access to a map has always more overhead than just access a constant value in scope.